code/parse/sexp.cpp

```diff
@@ -11029,7 +11029,9 @@ void sexp_change_argument_validity(int n, bool invalidate)
 		// first we must check if the arg_handler marks a selection. At the moment random-of is the only one that does this
 		arg_n = CDR(arg_handler);
 		while (invalidate && (arg_n != -1)) {
-			if (Sexp_nodes[arg_n].flags & SNF_ARGUMENT_SELECT) {
+			// (in)validating container arguments isn't supported
+			if ((Sexp_nodes[arg_n].subtype != SEXP_ATOM_CONTAINER_NAME) &&
```

Comment:
`SEXP_ATOM_CONTAINER_NAME` isn't supported, but `SEXP_ATOM_CONTAINER_DATA` is?

Reply:
No, it isn't supported, either.

Support for using Replace Container Data in argument list was removed from FRED in PR #4392 but I forgot then to add a check in `check_sexp_syntax()`. The new commit fixes that.

Also, it occurred to me that since the argument type for `(in)validate-argument` is `OPF_ANYTHING`, which excludes containers (see `check_sexp_syntax()`), an `Assertion()` makes more sense than an `if`-check, so that's now fixed, too.

Reply:
Ok.